### PR TITLE
Orphaned clusters removal tool

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Add the tool for deleting orphaned clusters

This pull request does not integrate the tool with any prow job, it will be done in a separate one.

**Related issue(s)**
See also #177

### CODE REVIEW

File: development/tools/cmd/clusterscollector/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+### Flags
+
+See the list of available flags:
+
+| Name                      | Required | Description                                                                                          |
+| :------------------------ | :------: | :--------------------------------------------------------------------------------------------------- |
+| **--project**             |   Yes    | GCP project name.
+| **--dryRun**              |    No    | The boolean value that controls the dry-run mode. It defaults to `true`.
+| **--ageInHours**          |    No    | The integer value for the number of hours. It only matches clusters older than `now()-ageInHours`. It defaults to `3`.
+| **--clusterNameRegexp**   |    No    | The string value with a valid Golang regexp. It is used to match clusters by their name. It defaults to `^gkeint[-](pr|commit)[-].*`.
```
Comment (strekm) [inline development/tools/cmd/clusterscollector/README.md]: to ignore regexp interpretation by milv tool please add the following configuration to `milv.config.yaml` file:
```
  - path: "path_to_this_file"
    config:
      white-list-internal: ["pr|commit"]
```

Comment (Tomasz-Smelcerz-SAP) [reply development/tools/cmd/clusterscollector/README.md]: Done.

File: development/tools/cmd/clusterscollector/main.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	"github.com/kyma-project/test-infra/development/tools/pkg/clusterscollector"
+	"github.com/kyma-project/test-infra/development/tools/pkg/common"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/oauth2/google"
+	compute "google.golang.org/api/compute/v1"
+	container "google.golang.org/api/container/v1"
+)
+
+const defaultClusterNameRegexp = "^gkeint[-](pr|commit)[-].*"
+const defaultJobLabelRegexp = "^kyma-gke-integration"
```
[the remark this answers is not in the document]
Comment (Tomasz-Smelcerz-SAP) [reply development/tools/cmd/clusterscollector/main.go]: Done.

File: development/tools/cmd/clusterscollector/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Clusters Garbage Collector
+
+## Overview
+
+This command finds and removes orphaned clusters created by the `kyma-gke-integration` job in a Google Cloud Platform (GCP) project.
+
+The `kyma-gke-integration` job creates a GKE cluster in order to install and test Kyma.
```
Comment (kazydek) [inline development/tools/cmd/clusterscollector/README.md]:
```suggestion
The `kyma-gke-integration` job creates a GKE cluster to install and test Kyma.
```

File: development/tools/cmd/clusterscollector/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+| **--clusterNameRegexp**   |    No    | The string value with a valid Golang regexp. It is used to match clusters by their name. It defaults to `^gkeint[-](pr|commit)[-].*`.
+| **--jobLabelRegexp**      |    No    | The string value with a valid Golang regexp. It is used to match clusters by the `job` label value. It defaults to `^kyma-gke-integration$`.
+
+### Environment variables
+
+See the list of available environment variables:
+
+| Name                                  | Required | Description                                                                                          |
+| :------------------------------------ | :------: | :--------------------------------------------------------------------------------------------------- |
+| **GOOGLE_APPLICATION_CREDENTIALS**    |    Yes   | The path to the service account file. The service account requires at least: `container.clusters.list` and `container.clusters.delete` Google IAM permissions. |
```
Comment (kazydek) [inline development/tools/cmd/clusterscollector/README.md]:
```suggestion
| **GOOGLE_APPLICATION_CREDENTIALS**    |    Yes   | The path to the service account file. The service account requires at least `container.clusters.list` and `container.clusters.delete` Google IAM permissions. |
```

File: development/tools/cmd/clusterscollector/README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    --dryRun=false
+```
+
+### Flags
+
+See the list of available flags:
+
+| Name                      | Required | Description                                                                                          |
+| :------------------------ | :------: | :--------------------------------------------------------------------------------------------------- |
+| **--project**             |   Yes    | GCP project name.
```
Comment (kazydek) [inline development/tools/cmd/clusterscollector/README.md]:
```suggestion
| **--project**             |   Yes    | GCP project name
```

File: development/tools/cmd/clusterscollector/README.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+| **--dryRun**              |    No    | The boolean value that controls the dry-run mode. It defaults to `true`.
```
Comment (kazydek) [inline development/tools/cmd/clusterscollector/README.md]:
```suggestion
| **--dryRun**              |    No    | The Boolean value that controls the dry-run mode. It defaults to `true`.
```

**Pull request merged**